application/frontend/controllers/UserController.php

```diff
@@ -53,6 +58,11 @@ public function actionUpdate(string $employeeId)
 
         $user->attributes = Yii::$app->request->getBodyParams();
 
+        /*
+         * Received as boolean, stored as tinyint(1)
+         */
+        $user->do_not_disclose = (int)$user->do_not_disclose;
```

Comment:
Is this type coercion one we have to do manually? I ask because I don't remember this being an issue before with boolean columns in the database.

Reply:
Unfortunately, yes. The type validation in Yii doesn't recognize `false` as an integer. Interestingly, `true` **is** recognized as an integer. I found the reason has to do with the way PHP casts `bool` to `string`. `true` becomes '1' and `false` becomes ''
I did notice that some other database columns use enums such as 'yes', 'no'. Perhaps this solution circumvents the problem.